source/includes/_ethereum.md

```diff
@@ -373,8 +415,56 @@ contract KrakenPriceTicker is usingOraclize {
 
 ```
 
-You have to consider that your account will be debited for most of your Oraclize calls. If your contract is not covered with enough ETH, the query will fail. Depending on your contract logic you may want to check the price for your next query before it gets send. You can do this by calling `oraclize_getPrice` and check if it is higher than your current contract balance. If that's the case the `oraclize_query` will fail and you may want to handle it gracefully. You can also add a gaslimit parameter to the `oraclize_getPrice` function: `oraclize_getPrice(string datasource, uint gaslimit)`. Make sure that the custom gaslimit for `oraclize_getPrice` matches with the one you will use for `oraclize_query`.
+Payment for Provable queries are debited directly from the contract that calls the `oraclize_query(...)'` function at the moment that function call is made. So - depending on your contract's logic - you may want to know ahead of time the price of your next query before making it. Provable provides a helper method to do just that: `oraclize_getPrice(string _datasource);`.
+
```

Comment:
regarding the piggies article, it may be worth to maybe point out 'atomic-like' support, where the contract balance is empty, and needs to be funded by users each time (effectively how etheroll workds too)

Reply:
Yeah good point - super timely in fact. Will update to reflect that.